index.js

```diff
@@ -99,6 +99,7 @@ ZongJi.prototype._isChecksumEnabled = function(next) {
         // MySQL < 5.6.2 does not support @@GLOBAL.binlog_checksum
         return next(false);
       }
+      // TODO: Remove throws to make the module safe
```

Comment:
Constructors are safe to call and are considered ready to go once they return a value.

The connection is a property of the object, not the object itself. I feel that the connection should not dictate the lifecycle of the object (or a process) the developer should.

If we're going to allow/encourage the developer to implement their own connection handling, it probably shouldn't be in the constructor. Sure, you could do it, but it will be deeply nested and very clunky.